### PR TITLE
Refactor Release model to include ReleasedFile

### PR DESCRIPTION
**Pull request by bloodearnest**
Each file now has its own ReleasedFile model, which makes it easier to
track in future. The API is unchanged. It also moves the release files
under workspace/releases/id, to leave space for workspace/published, etc
in future.

Also realised my ReleaseFactory had reimplemented `handle_release()`
logic, so just called `handle_release()` to create the Release for
consistency.

I discovered and fixed a bug on the way - the `manifest.json` was being
included in the id hash. This meant if the manifest had changed, but the
files had not, it would hash differently, which was not the intended
goal. And we may switch to sending manifests via job-runner in future,
so it seemed to make sense to exclude it.

The migration is destructive, but we have no data in prod anyway.

### CODE REVIEW

**Review by ghickman — APPROVED** [review]
Some naming changes (mostly for consistency), otherwise this looks great.

The ReleaseFactory implementation continues to be great too 👍

File: jobserver/models/outputs.py [hunk 1]
```diff
@@ -30,10 +36,8 @@ class Release(models.Model):
         on_delete=models.PROTECT,
         related_name="+",
     )
-    published_at = models.DateTimeField(default=timezone.now)
+    released_at = models.DateTimeField(default=datetime.utcnow)
```
[the remark this answers is not in the document]
Comment (bloodearnest) [reply jobserver/models/outputs.py:39]: sure

File: jobserver/models/outputs.py [hunk 2]
```diff
@@ -14,7 +20,7 @@ class Release(models.Model):
     id = models.TextField(primary_key=True)
     # TODO: link this formally via backend_username once we have a mapping
     # between User and their backend username
-    publishing_user = models.ForeignKey(
+    releasing_user = models.ForeignKey(
```
[the remark this answers is not in the document]
Comment (bloodearnest) [reply jobserver/models/outputs.py:23]: ok

**Comment by bloodearnest** [comment]
I've applied your feedback, but I didn't want to commits with different versions of the migration, so I pre-squashed all the commits into one.

**Pull request merged**